src/cpp/src/sequence_group.hpp

```diff
@@ -475,8 +479,11 @@ class SequenceGroup {
         } else if (has_finished()) {
             set_generation_status(GenerationStatus::FINISHED);
         }
+        if (handle_dropped()) {
+            push_empty();
+        }
```

Comment:
Perhaps we could move this up to the beginning of this method and return early from this block?
If handle is dropped we don't have to check anything else.

Reply:
Outdated